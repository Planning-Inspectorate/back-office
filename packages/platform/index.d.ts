import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
