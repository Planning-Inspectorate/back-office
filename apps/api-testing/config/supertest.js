// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
