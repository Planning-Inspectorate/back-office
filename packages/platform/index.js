export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
