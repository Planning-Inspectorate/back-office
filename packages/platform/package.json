{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
