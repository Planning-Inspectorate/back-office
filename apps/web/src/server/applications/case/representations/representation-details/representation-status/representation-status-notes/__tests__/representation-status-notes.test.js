import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../../../testing/index.js';
import { representationDetailsFixture } from '../../../__fixtures__/representation-details.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };
const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
	nock('http://test/')
		.get(`/applications/1/representations/1`)
		.reply(200, representationDetailsFixture);

	nock('http://test/')
		.patch(`/applications/1/representations/1`, {
			status: 'INVALID',
			invalidReason: 'Duplicate',
			updatedBy: 'Joe Blogs'
		})
		.reply(200, { message: 'all good' });
};

describe('Change representation status page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl =
		'/applications-service/case/1/relevant-representations/1/representation-details/status-result';

	describe('GET /applications-service/case/1/relevant-representations/1/representations-details/status-result', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/case-team');
		});

		it('should render the page', async () => {
			const response = await request.get(`${baseUrl}?changeStatus=AWAITING_REVIEW`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should contain specified radio options is status is REFERRED', async () => {
			const response = await request.get(`${baseUrl}?changeStatus=REFERRED`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toContain('Case Team');
			expect(element.innerHTML).toContain('Inspector');
			expect(element.innerHTML).toContain('Central Admin Team');
			expect(element.innerHTML).toContain('Interested Party');
		});

		it('should contain specified radio options if status is INVALID', async () => {
			const response = await request.get(`${baseUrl}?changeStatus=INVALID`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toContain('Duplicate');
			expect(element.innerHTML).toContain('Merged');
			expect(element.innerHTML).toContain('Not relevant');
			expect(element.innerHTML).toContain('Resubmitted');
			expect(element.innerHTML).toContain('TEST');
		});
	});

	describe('POST /applications-service/case/1/relevant-representations/1/representations-details/status-result', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/case-team');
		});

		it('should show validation error if no option was selected (except for WITHDRAWN and AWAITING_REVIEW)', async () => {
			const response = await request.post(`${baseUrl}?changeStatus=INVALID`).send({});
			const element = parseHtml(response.text);

			expect(element.innerHTML).toContain('Select one option');
		});
	});
});
