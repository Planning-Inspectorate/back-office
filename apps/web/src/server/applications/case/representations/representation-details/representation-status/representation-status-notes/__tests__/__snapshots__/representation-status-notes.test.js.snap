// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change representation status page GET /applications-service/case/1/relevant-representations/1/representations-details/status-result should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\"><span class=\\"govuk-caption-l\\"></span>
    <h1 class=\\"govuk-heading-l govuk-!-margin-bottom-2\\">Notes</h1><strong class=\\"govuk-heading-s govuk-!-margin-bottom-6\\"><strong class=\\"govuk-tag govuk-tag--grey\\">awaiting review</strong></strong>
    <div     class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <form method=\\"post\\" novalidate>
                <div class=\\"govuk-form-group\\">
                    <fieldset class=\\"govuk-fieldset\\">
                        <div class=\\"govuk-radios\\" data-module=\\"govuk-radios\\"></div>
                    </fieldset>
                </div>
                <div class=\\"govuk-form-group\\">
                    <div id=\\"notes-hint\\" class=\\"govuk-hint\\">Use this space to provide more details.</div>
                    <textarea class=\\"govuk-textarea\\"
                    id=\\"notes\\" name=\\"notes\\" rows=\\"5\\" aria-describedby=\\"notes-hint\\"></textarea>
                </div>
                <button type=\\"submit\\" class=\\"govuk-button govuk-!-margin-top-3\\" data-module=\\"govuk-button\\">Save changes</button>
            </form>
        </div>
        </div>
</main>"
`;
